aodndata/acorn/handler.py

```diff
@@ -0,0 +1,106 @@
+            acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+    elif "FV00_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+    elif "FV01_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_QC"
+    else:
+        raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                   format(filename=os.path.basename(filepath)))
+
+    return acorn_file_type
+
+
+class AcornHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(AcornHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc']
+
+    def preprocess(self):
+        nc_collection = self.file_collection[0]
+
+        # enforce a check on the include_regexes and the filename to make sure the handler is processing the correct
+        # matching rexep file written in the config file
+        m = re.search(self.include_regexes[0], nc_collection._name)
+        if m is None:
+            raise InvalidFileNameError(
+                "file name: \"{filename}\" not allowed in this pipeline".format(
+                    filename=os.path.basename(nc_collection._name)))
+
+        # if file contains any of FILE_TYPE_NEED_INDEX, we index, otherwise publish only
+        if any(s in get_type(nc_collection._name) for s in FILE_TYPE_NEED_INDEX):
+            nc_collection.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+        else:
+            nc_collection.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+
+        # check if file with same dest path already on s3. If yes, check it's date_created nc attribute to know
+        # if we need to overwrite this object or not
+        #     destination_s3 = self.dest_path(nc_collection._name)
+
+        #nc_collection.dest_path = os.path.join(destination_s3, nc_collection.name)
+        #storage_query_res = self.state_query.query_storage(destination_s3)
+        #if storage_query_res:
+        #    creation_date_nc_pipeline = self.get_creation_date(nc_collection)  # creation date of the file in the pipeline
+
+        #    #TODO get leigh to create the funcion below
+        #    nc_s3_path = download_s3(destination_s3)
+        #    creation_date_nc_s3 = get_creation_date(nc_s3_path)  # creation date of the file on s3
+        #
+        #    if creation_date_nc_pipeline > creation_date_nc_s3:
+        #        nc_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+    @staticmethod
+    def get_creation_date(netcdf_path):
+        with Dataset(netcdf_path, mode='r') as nc_obj:
+            nc_pipeline_date_created_str = nc_obj.date_created
+            nc_pipeline_date_created = datetime.strptime(nc_pipeline_date_created_str, '%Y-%m-%dT%H:%M:%SZ')
+        return nc_pipeline_date_created
+
+    @staticmethod
+    def dest_path(filepath):
+        m = re.search(r'^IMOS_ACORN_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_([A-Z]{3,4})_(FV0[0-1])_(.*)\.nc$',
```

Comment:
I think named groups are really good for when there is a strict naming convention like this. It makes the code later on much more readable than having to later map a numeric index to one of the fields in the file name.

For example, changing nothing else other than adding the named fields:

```python
    @staticmethod
    def dest_path(filepath):
        pattern = re.compile(r"""
                        ^IMOS_ACORN_
                        (?P<named_field1>[A-Z].*)_
                        (?P<nc_time_cov_start>[0-9]{8}T[0-9]{6}Z)_
                        (?P<platform_code>[A-Z]{3,4})_
                        (?P<another_field>FV0[0-1])_
                        (?P<yet_another_field>.*)
                        \.nc$
                        """, re.VERBOSE)

        m = pattern.match(os.path.basename(filepath))
        if m is None:
            raise InvalidFileNameError(
                "file name: \"{filename}\" not matching regex to deduce dest_path for ACORN".format(
                    filename=os.path.basename(filepath)))

        fields = m.groupdict()

        nc_time_cov_start = datetime.strptime(fields['nc_time_cov_start'], '%Y%m%dT%H%M%SZ')

        file_type = get_type(filepath)
        return os.path.join('IMOS', 'ACORN', file_type, fields['platform_code'],
                            '%d' % nc_time_cov_start.year, '%02d' % nc_time_cov_start.month,
                            '%02d' % nc_time_cov_start.day,
                            os.path.basename(filepath)
                            )
```

Reply:
thanks I didn't know how to do that

Reply:
@lbesnard can you please update your code following @lwgordonimos 's recommendations above?
Fields are data_codes, time_coverage_start, platform_code, file_version and file_type_code.

Reply:
yeap already done, but haven't pushed it yet